pypeit/extraction.py

```diff
@@ -391,25 +392,29 @@ def run(self, global_sky, prepare_extraction=True, model_noise=None, spat_pix=No
 
         self.global_sky = global_sky
 
+        # Apply a global flexure correction to each slit
+        # provided it's not a standard star
+        if self.par['flexure']['spec_method'] != 'skip' and not self.std_redux:
+            self.spec_flexure_correct(mode='global')
+
+        # If flexure correction failed and `excessive_shift == skip`, the
+        #   specified shift will be NaN.  Skip object extraction by setting
+        #   self.sobjs_obj to a blank container (i.e., self.nobj_to_extract = 0)
+        if np.logical_not(np.isfinite(self.slitshift)).any():
+            self.sobjs_obj = specobjs.SpecObjs()
```

Comment:
I don't think it's necessary to skip the extraction of all the objects. Especially for multislit spectroscopy it's more common to have the flexure correction to fail for only one or a few slits. So, instead of setting `self.sobjs_objs` to a black container, we can just remove the object (using the method `remove_sobj`, there are some examples above) that is in the slit with bad/failed flexure correction. 

I also want to mention that below a local flexure correction is applied (i.e., the global correction is fine tuned for each specific extracted object) and it may happen that the flexure correction fails there too.

Reply:
The scope of changes needed for skipping extraction is getting beyond my level of expertise.  I have therefore reverted this last change (there is no more `skip_extraction` option) and renamed `skip` --> `set_to_zero`.

The machinery is in place now for someone more skilled to add a `skip_extraction` option for `excessive_shift` if desired.